src/dyad/modules/dyad.c

```diff
@@ -164,21 +185,32 @@ dyad_fetch_request_cb (flux_t *h, flux_msg_handler_t *w, const flux_msg_t *msg,
     }
 #endif  // DYAD_SPIN_WAIT
 
+    DYAD_PERF_REGION_BEGIN (ctx->perf_handle, "dyad_server_read_data");
     FLUX_LOG_INFO (h, "Reading file %s for transfer", fullpath);
     fd = open (fullpath, O_RDONLY);
     if (fd < 0) {
         FLUX_LOG_ERR (h, "DYAD_MOD: Failed to open file \"%s\".\n", fullpath);
+        DYAD_PERF_REGION_END (ctx->perf_handle, "dyad_server_read_data");
         goto fetch_error;
     }
-    if ((inlen = read_all (fd, &inbuf)) < 0) {
+    FLUX_LOG_INFO (h, "Getting file size");
+    file_size = get_file_size (fd);
+    FLUX_LOG_INFO (h, "Getting buffer for reading the file");
+    rc = ctx->dtl_handle->get_buffer (ctx->dtl_handle, file_size, &inbuf);
+    FLUX_LOG_INFO (h, "Reading file data into buffer");
+    if ((inlen = read (fd, inbuf, file_size)) < 0) {
```

Comment:
I change the call to your version of `read_all()` to `read()`. So, there is no need to change `read_all.{h,c}`
In fact, the error semantic of the function was incorrect.